### PR TITLE
Add Mirrored flag

### PR DESCRIPTION
**Pull request by wmmiii**
This will allow us to have chained strips that display the same pattern
on each, regardless of orientation.

Also add helper function to mirror indicies.

### CODE REVIEW

File: software/generic/test/MathTest.cpp [hunk 1]
```diff
@@ -55,3 +55,55 @@ TEST(Math, shouldGetCardinalCordinates) {
   ASSERT_EQ(angle, 223);
   ASSERT_EQ(radius, 40);
 }
+
+TEST(Math, shouldMirrorIndiciesOdd) {
+  uint8_t led_index;
+  uint8_t led_count;
+
+  led_index = 0;
```
Comment (ademuri) [inline software/generic/test/MathTest.cpp]: Optional: do you want to make these data driven tests?

**Review by ademuri — APPROVED** [review]

File: software/generic/test/MathTest.cpp (hunk 1 above)
Comment (wmmiii) [reply software/generic/test/MathTest.cpp]: Sure, do you have an example of one I can follow?

Comment (wmmiii) [reply software/generic/test/MathTest.cpp]: Or do I just write the loop myself?

Comment (ademuri) [reply software/generic/test/MathTest.cpp]: I don't have a good example of one - you just write the loop yourself, and use expect. It's a good idea to stream useful error messages to expect, too.

Comment (wmmiii) [reply software/generic/test/MathTest.cpp]: Done.

**Pull request merged**